plugins/catalog-backend/src/service/DefaultEntitiesCatalog.test.ts

```diff
@@ -109,12 +109,23 @@ describe('DefaultEntitiesCatalog', () => {
     return Promise.all(
       Object.entries(previousValue).map(async ([key, value]) => {
         const currentKey = `${previousKey ? `${previousKey}.` : ``}${key}`;
-        if (typeof value === 'object') {
+
+        if (typeof value === 'object' && !Array.isArray(value)) {
           await insertSearchRow(knex, id, currentKey, value);
+        } else if (Array.isArray(value)) {
```

Comment:
This seems like a separate thing. Is it a bug fix?

Reply:
Yep. While inserting the search row for `metadata.tags: ['java, node']`, since Array is considered as a object, it ends up getting inserted in the search table as:
      
```
       {
          entity_id: '026c3d12-677d-4602-ba79-12f01fc80d2f',
          key: 'metadata.tags.0',
          original_value: 'java',
          value: 'java'
        },
        {
          entity_id: '026c3d12-677d-4602-ba79-12f01fc80d2f',
          key: 'metadata.tags.1',
          original_value: 'node',
          value: 'node'
        }
```
I had to handle the case for Arrays separately to make the tests pass.

Reply:
Ah this code is unfortunate and duplicates what the ACTUAL search builder does. So you could remove this function and instead add this at the bottom of `addEntityToSearch`:

```ts
    for (const row of buildEntitySearch(id, entity)) {
      await knex<DbSearchRow>('search').insert({
        entity_id: id,
        key: row.key,
        value: row.value,
        original_value: row.original_value,
      });
    }
```

(`buildEntitySearch` is imported from the nearby stitching folder)